app/src/main/java/org/groundplatform/android/ui/settings/LocaleHelper.kt

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright 2025 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.groundplatform.android.ui.settings
+
+import android.content.Context
+import android.content.res.Configuration
+import androidx.preference.PreferenceManager
+import java.util.Locale
+
+fun onAttach(context: Context, defaultLanguage: String = "en"): Context {
```

Comment:
Can we improve the method name to be more appropriate? This function doesn't need to know if it is called during a lifecycle event. Instead the name should reflect what the function is doing. Also, please add some ktdoc to it.